src/framework/ui/view/interactiveprovider.cpp

```diff
@@ -592,7 +592,7 @@ RetVal<InteractiveProvider::OpenData> InteractiveProvider::openWidgetDialog(cons
 
     bool sync = q.param("sync", Val(false)).toBool();
     if (sync) {
-        dialog->exec();
+        dialog->open();
```

Comment:
If the `sync` parameter is set, then it is impossible to call `open` here - it is asynchronous, here it is expected that a synchronous call is expected to return the result